source/common/filesystem/subscription_impl.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/filesystem/filesystem.h"
+
+#include "common/common/macros.h"
+#include "common/config/utility.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include "api/base.pb.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+/**
+ * Filesystem inotify implementation of the API Subscription interface. This allows the API to be
+ * consumed on filesystem changes to files containing the JSON canonical representation of
+ * lists of ResourceType.
+ */
+template <class ResourceType> class SubscriptionImpl : public Config::Subscription<ResourceType> {
```

Comment:
I've placed the `Subscription` interface in the `Config` namespace, then each of the `SubscriptionImpl` subclasses in `Grpc`, `Http`, `Filesystem` as needed. Are you suggesting a `Config` namespace within these?